substrate/primitives/staking/src/lib.rs

```diff
@@ -417,18 +425,28 @@ impl<
 }
 
 /// A snapshot of the stake backing a single validator in the system.
-#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
-pub struct ExposurePage<AccountId, Balance: HasCompact> {
+#[derive(
+	PartialEq, Eq, PartialOrd, Ord, Clone, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen,
+)]
+#[scale_info(skip_type_params(MaxExposurePageSize))]
+pub struct ExposurePage<
+	AccountId,
+	Balance: HasCompact + MaxEncodedLen,
+	MaxExposurePageSize: Get<u32>,
+> {
 	/// The total balance of this chunk/page.
 	#[codec(compact)]
 	pub page_total: Balance,
 	/// The portions of nominators stashes that are exposed.
-	pub others: Vec<IndividualExposure<AccountId, Balance>>,
+	pub others: WeakBoundedVec<IndividualExposure<AccountId, Balance>, MaxExposurePageSize>,
 }
 
-impl<A, B: Default + HasCompact> Default for ExposurePage<A, B> {
+impl<A, B: Default + HasCompact + MaxEncodedLen, C: Get<u32>> Default for ExposurePage<A, B, C> {
 	fn default() -> Self {
-		ExposurePage { page_total: Default::default(), others: vec![] }
+		ExposurePage {
+			page_total: Default::default(),
+			others: WeakBoundedVec::force_from(vec![], None),
```

Comment:
if you want you can create a `fn new` method on `WeakBoundedVec`

Reply:
Yes, I just opened this PR

https://github.com/paritytech/parity-common/pull/888

Reply:
Just call `default`.

Reply:
Ok, changed to `default()` in https://github.com/paritytech/polkadot-sdk/pull/6445/commits/4b58f0ac69cf465a8c5616cae3fe9dacee047b09